### PR TITLE
✨ aws ec2 eips, vpc natgateways, vpc peering conn, service endpoints

### PR DESCRIPTION
**Pull request by vjeffrey**

### CODE REVIEW

File: providers/aws/resources/aws.lr [hunk 1]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  eips() []aws.ec2.eip
+}
+
+// Amazon Elastic IP
+private aws.ec2.eip {
+  // PublicIP assigned to 
+  publicIp string
+  // false if allocationId not present
+  attached bool 
+  // Ec2 instance associated with the EIP
```
Comment (czunker) [inline providers/aws/resources/aws.lr:2276]: I'm not that familiar with AWS. Can an EIP only be attached to an EC2 instance. Not something else?

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 2]
```diff
@@ -898,6 +898,8 @@ resources:
         Use the `aws.ec2` resource to assess the configuration of AWS EC2 instances.
     fields:
       ebsEncryptionByDefault: {}
+      eips:
+        min_mondoo_version: 9.0.0
```
Comment (czunker) [inline providers/aws/resources/aws.lr.manifest.yaml:902]: Shouldn't this be 11+?

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 3]
```diff
@@ -940,6 +942,22 @@ resources:
...
+      instance: {}
+      networkInterfaceId: {}
+      networkInterfaceOwnerId: {}
+      privateIpAddress: {}
+      publicIp: {}
+      publicIpv4Pool: {}
+      region: {}
+      tags: {}
+    is_private: true
+    min_mondoo_version: 9.0.0
```
Comment (czunker) [inline providers/aws/resources/aws.lr.manifest.yaml:957]: Same here and on the other resources

File: providers/aws/resources/aws_vpc.go [hunk 4]
```diff
@@ -102,6 +104,128 @@ func (a *mqlAws) getVpcs(conn *connection.AwsConnection) []*jobpool.Job {
 	return tasks
 }
 
+func (a *mqlAwsVpcNatgatewayAddress) id() (string, error) {
+	return a.AllocationId.Data, nil
```
Comment (czunker) [inline providers/aws/resources/aws_vpc.go:108]: Can a NAT GW IP not be assigned and this empty?

**Review by czunker — COMMENTED** [review]
Nice list of resources

File: providers/aws/resources/aws.lr.manifest.yaml (hunk 2 above)
Comment (vjeffrey) [reply providers/aws/resources/aws.lr.manifest.yaml:902]: bc providers auto update / aren't tied to the version, anything after 9 should work

**Review by czunker — APPROVED** [review]
Thanks @vjeffrey

File: providers/aws/resources/aws.lr.manifest.yaml (hunk 2 above)
Comment (czunker) [reply providers/aws/resources/aws.lr.manifest.yaml:902]: Ah, good to know. I wasn't aware of this. Thanks.

File: providers/aws/resources/aws.lr [hunk 5]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
   networkAcls() []aws.ec2.networkacl
   // List of keypairs for the account
   keypairs() []aws.ec2.keypair
+  // List of EIPs
+  eips() []aws.ec2.eip
+}
+
+// Amazon Elastic IP
+private aws.ec2.eip {
+  // PublicIP assigned to
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Public IP address of the EIP
```

File: providers/aws/resources/aws.lr [hunk 6]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
   keypairs() []aws.ec2.keypair
+  // List of EIPs
+  eips() []aws.ec2.eip
+}
+
+// Amazon Elastic IP
+private aws.ec2.eip {
+  // PublicIP assigned to 
+  publicIp string
+  // false if allocationId not present
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Whether the Elastic IP is associated with an instance (false if no allocationId is present)
```

File: providers/aws/resources/aws.lr [hunk 7]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
   networkAcls() []aws.ec2.networkacl
   // List of keypairs for the account
   keypairs() []aws.ec2.keypair
+  // List of EIPs
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // List of Elastic IPs (EIPs)
```

File: providers/aws/resources/aws.lr [hunk 8]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  eips() []aws.ec2.eip
+}
+
+// Amazon Elastic IP
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
// Amazon Elastic IP (EIP)
```

File: providers/aws/resources/aws.lr [hunk 9]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+private aws.ec2.eip {
+  // PublicIP assigned to 
+  publicIp string
+  // false if allocationId not present
+  attached bool 
+  // Ec2 instance associated with the EIP
+  instance() aws.ec2.instance
+  // Id for the network interface
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // ID of the network interface
```

File: providers/aws/resources/aws.lr [hunk 10]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  networkInterfaceId string
+  // Id for the network interface owner
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // ID of the network interface owner
```

File: providers/aws/resources/aws.lr [hunk 11]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  networkInterfaceOwnerId string
+  // Private IP address for the EIP
+  privateIpAddress string
+  // Ipv4 pool for the EIP
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // IPv4 pool of the EIP
```

File: providers/aws/resources/aws.lr [hunk 12]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  tags map[string]string
+  // Region where the EIP is located
+  region string
+}
+
+// Amazon VPC NAT Gateway
+private aws.vpc.natgateway {
+  // Time when the gateway was created
+  createdAt time 
+  // Id for the NAT gateway
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // ID of the NAT gateway
```

File: providers/aws/resources/aws.lr [hunk 13]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  tags map[string]string
+  // VPC associated with the NAT gateway
+  vpc() aws.vpc
+  // List of addresses associated with the NAT gateway
+  addresses []aws.vpc.natgateway.address
+}
+
+// Amazon VPC NAT gateway address
+private aws.vpc.natgateway.address {
+  // Allocation Id for the address
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Allocation ID for the address
```

File: providers/aws/resources/aws.lr [hunk 14]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  allocationId string
+  // Network interface Id for the address
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Network interface ID for the address
```

File: providers/aws/resources/aws.lr [hunk 15]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  networkInterfaceId string
+  // Private IP associated with the address
+  privateIp string
+  // EIP associated with the address
+  publicIp() aws.ec2.eip // AllocationId can get us back to the actual EIP
+  // True if address is the primary one for the NAT gateway
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Whether this is the primary address for the NAT gateway
```

File: providers/aws/resources/aws.lr [hunk 16]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  isPrimary bool
+}
+
+// Amazon VPC Service Endpoint
+private aws.vpc.serviceEndpoint {
+  // True if acceptance required
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr:2324]:
```suggestion
  // Whether acceptance is required
```
I'd love to finish this thought -- required for for what to happen with this endpoint?

File: providers/aws/resources/aws.lr [hunk 17]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  acceptanceRequired bool
+  // List of availability zones for the service endpoint
+  availabilityZones []string
+  // List of base endpoint DNS names for the service endpoint
+  dnsNames []string  
+  // Service Id
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Service ID
```

File: providers/aws/resources/aws.lr [hunk 18]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  id string
+  // True if service endpoint manages VPC endpoints
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Whether the service endpoint manages VPC endpoints
```

File: providers/aws/resources/aws.lr [hunk 19]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  payerResponsibility string
+  // Service private DNS name verification state
+  privateDnsNameVerificationState string
+  // List of service private DNS names
+  privateDnsNames []string
+  // Tags for the service endpoint
+  tags map[string]string
+  // Service type
+  type string
+  // True if service supports VPC endpoint policy
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Whether the service supports VPC endpoint policy
```

File: providers/aws/resources/aws.lr [hunk 20]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  vpcEndpointPolicySupported bool
+}
+
+// Amazon VPC Peering Connection
+private aws.vpc.peeringConnection {
+  // VPC for the peering connection acceptor
+  acceptorVpc() aws.vpc.peeringConnection.peeringVpc
+  // Expiration time for the peering connection
+  expirationTime time
+  // Id for the peering connection
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // ID of the peering connection
```

File: providers/aws/resources/aws.lr [hunk 21]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  requestorVpc() aws.vpc.peeringConnection.peeringVpc
+  // Status of the peering connection
+  status string
+  // Tags for the peering connection
+  tags map[string]string
+}
+
+// Amazon VPC Peering Connection Peering VPC
+private aws.vpc.peeringConnection.peeringVpc {
+  // True if DNS resolution from remote VPC allowed
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Whether DNS resolution from the remote VPC is allowed
```

File: providers/aws/resources/aws.lr [hunk 22]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  allowDnsResolutionFromRemoteVpc bool
+  // True if egress allowed
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Whether egress is allowed from a local classic link to the remote VPC
```

File: providers/aws/resources/aws.lr [hunk 23]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  allowEgressFromLocalClassicLinkToRemoteVpc bool
+  // True if egress allowed
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Whether egress is allowed from a local VPC to a classic link
```

File: providers/aws/resources/aws.lr [hunk 24]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  allowEgressFromLocalVpcToRemoteClassicLink bool
+  // List of ipv4 CIDR blocks for peering connection
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // List of IPv4 CIDR blocks for peering connection
```

File: providers/aws/resources/aws.lr [hunk 25]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  ipv4CiderBlocks []string 
+  // List of ipv6 CIDR blocks for peering connection
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // List of IPv6 CIDR blocks for peering connection
```

File: providers/aws/resources/aws.lr [hunk 26]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  ipv6CiderBlocks []string
+  // Owner Id for the peering connection
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Owner ID of the peering connection 
```

File: providers/aws/resources/aws.lr [hunk 27]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  allowEgressFromLocalClassicLinkToRemoteVpc bool
+  // True if egress allowed
+  allowEgressFromLocalVpcToRemoteClassicLink bool
+  // List of ipv4 CIDR blocks for peering connection
+  ipv4CiderBlocks []string 
+  // List of ipv6 CIDR blocks for peering connection
+  ipv6CiderBlocks []string
+  // Owner Id for the peering connection 
+  ownerID string
+  // Region for the peering connection
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Region of the peering connection
```

File: providers/aws/resources/aws.lr [hunk 28]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  region string
+  // VPC associated with the peering connection, populated if in same account
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // VPC associated with the peering connection (populated if it's in the same account)
```

File: providers/aws/resources/aws.lr [hunk 29]
```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
...
+  vpc() aws.vpc
+  // Id of VPC associated with the peering connection
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // ID of the VPC associated with the peering connection
```

**Review by misterpantz — COMMENTED** [review]
Hi VJ! Thank you for thoroughly documenting these new resources! Tim asked me to go through and polish the help. Please let me know if any of my suggestions don't make sense.

File: providers/aws/resources/aws_vpc.go (hunk 4 above)
Comment (vjeffrey) [reply providers/aws/resources/aws_vpc.go:108]: oh, yup, nice catch, modifying this

File: providers/aws/resources/aws.lr (hunk 1 above)
Comment (vjeffrey) [reply providers/aws/resources/aws.lr:2276]: seems like it from looking at the api

File: providers/aws/resources/aws.lr (hunk 16 above)
Comment (vjeffrey) [reply providers/aws/resources/aws.lr:2324]: from the api
```
	// Indicates whether VPC endpoint connection requests to the service must be
	// accepted by the service owner.
```

**Pull request merged**